From 60e597fdb5963229f4a78d55f219eefb1e4759e1 Mon Sep 17 00:00:00 2001
From: Srinu Jella <sjella@codeaurora.org>
Date: Thu, 23 Jan 2014 18:38:44 +0530
Subject: [PATCH 2/3] Bluetooth:GAP: Handle invalid SDP service search request

Handled the invalid SDP service search request by responding
with SDP_INVALID_REQ_SYNTAX.

Change-Id: Ide3e28bf552f75a915acc3dbb24fcee0b4cce9bb
CRs-Fixed: 598243
---
 stack/sdp/sdp_server.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/stack/sdp/sdp_server.c b/stack/sdp/sdp_server.c
index 92b9d16..6cf1896 100644
--- a/stack/sdp/sdp_server.c
+++ b/stack/sdp/sdp_server.c
@@ -98,6 +98,10 @@ static void process_service_search_attr_req (tCONN_CB *p_ccb, UINT16 trans_num,
 #define SDP_TEXT_BAD_CONT_INX   NULL
 #endif
 
+#ifndef SDP_TEXT_BAD_MAX_RECORDS_LIST
+#define SDP_TEXT_BAD_MAX_RECORDS_LIST   NULL
+#endif
+
 /*******************************************************************************
 **
 ** Function         sdp_server_handle_client_req
@@ -194,6 +198,14 @@ static void process_service_search (tCONN_CB *p_ccb, UINT16 trans_num,
     if (max_replies > SDP_MAX_RECORDS)
         max_replies = SDP_MAX_RECORDS;
 
+
+    if ((!p_req) || (p_req > p_req_end))
+    {
+        sdpu_build_n_send_error (p_ccb, trans_num, SDP_INVALID_REQ_SYNTAX, SDP_TEXT_BAD_MAX_RECORDS_LIST);
+        return;
+    }
+
+
     /* Get a list of handles that match the UUIDs given to us */
     for (num_rsp_handles = 0; num_rsp_handles < max_replies; )
     {
-- 
2.7.4

